contracts/src/v0.8/automation/dev/v2_2/AutomationRegistryLogicA2_2.sol

```diff
@@ -180,6 +180,24 @@ contract AutomationRegistryLogicA2_2 is AutomationRegistryBase2_2, Chainable {
     return executeCallback(id, payload);
   }
 
+  /**
+   * @dev checkCallback is used specifically for automation data streams lookups (see StreamsLookupCompatibleInterface.sol)
+   * @param id the upkeepID to execute a callback for
+   * @param values the values returned from the data streams lookup
+   * @param extraData the user-provided extra context data
+   */
+  function checkErrorHandler(
```

Comment:
Hey @infiloop2 I know we talked about adding the encoding (L#197) to AutomationUtils and have the offchain code call 1) the aforementioned encoding and 2) executeCallback as in L#207.  However I want to hear your thoughts on this proposal, I feel this one is more aligned with our current structure. 

1) we add this checkErrorHandler() in registry which is essentially doing the encoding. It will invoke the executeCallback afterward.
2) As in StreamsLookupInterface (contracts/src/v0.8/automation/interfaces/StreamsLookupCompatibleInterface.sol), we expect consumer contracts have both checkCallback and checkErrorHandler(). Note this is a shared interface btw 2.1 and 2.2, may need to consider compatiability.
3) Depends on the returned mercury reports, our offchain code can invoke either checkCallback (L171) or checkErrorHandler (L189) in registry.